exist-core/src/main/java/org/exist/xmlrpc/RpcConnection.java

```diff
@@ -1366,7 +1366,10 @@ private boolean parse(final byte[] xml, final XmldbURI docUri,
 
                 final long startTime = System.currentTimeMillis();
 
-                final MimeType mime = MimeTable.getInstance().getContentTypeFor(docUri.lastSegment());
+                MimeType mime = MimeTable.getInstance().getContentTypeFor(docUri.lastSegment());
```

Comment:
I think it would be better to add an additional media type parameter to the XML-RPC API for parse (in a backward compatible fashion) so that the Media Type is sent correctly from the XML-DB RemoteCollection class.

Reply:
@adamretter  I do not see the _backward_ part here as the content type seems to be added with 6.x.x and was not available prior. Either way those method will only invoked with XML content from the `RemoteCollection`..

Reply:
> I do not see the backward part here

By backward compatible, I mean that if we add a parameter then we effectively have an additional procedure available for RPC, so we must also keep the previous procedure as well. For example, at present there are 6 `parse` procedures available for RPC, we should add a 7th rather than remove/change an existing one.

> Either way those method will only invoked with XML content from the RemoteCollection

Sure, but by hardcoding the Media Type to just `application/xml` in the RPC procedure, we limit ourselves to just that. Instead the Media Type should be a procedure parameter that is passed in from the RPC client.

> where should the existing content type being taken on the RemoteCollection

It comes from the resource, i.e. `EXistResource#getMomeType()`.